engines/sci/graphics/palette.cpp

```diff
@@ -865,6 +860,31 @@ void GfxPalette::palVaryUpdate() {
 	}
 }
 
+void GfxPalette::delayForPalVaryWorkaround() {
+	if (_palVaryResourceId == -1)
+		return;
+	if (_palVaryPaused)
+		return;
+
+	// This gets called at the very beginning of kAnimate.
+	// If a zero-tick palVary is running, we delay briefly to give the
+	// palVary time to trigger.
+	//
+	// This prevents a race condition in Freddy Pharkas during nighttime,
+	// since we load pictures much faster (bug #5298).
+
+	if (_palVaryZeroTick) {
```

Comment:
I'm changing this slightly to emit a `warning()` if the flag still isn't toggled after 4 ticks. Maybe it's overly defensive, but since it's a workaround I'd prefer it to be robust and not cause hangs if we're misinterpreting the situation.